website/addons/github/static/githubNodeConfig.js

```diff
@@ -0,0 +1,456 @@
+'use strict';
+
+var ko = require('knockout');
+require('knockout.punches');
+var $ = require('jquery');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var $osf = require('js/osfHelpers');
+
+ko.punches.enableAll();
+
+var noop = function() {};
+
+var ViewModel = function(url, selector) {
+    var self = this;
+
+    self.accounts = ko.observable([]);
+
+    self.url = url;
+    self.selector = selector;
+
+    self.nodeHasAuth = ko.observable(false);
+    self.userHasAuth = ko.observable(false);
+    self.userIsOwner = ko.observable(false);
+    self.ownerName = ko.observable('');
+
+    self.urls = ko.observable({});
+    self.loadedSettings = ko.observable(false);
+    self.repoList = ko.observableArray([]);
+    self.loadedRepoList = ko.observable(false);
+    self.currentRepo = ko.observable('');
+    self.selectedRepo = ko.observable('');
+
+    self.accessToken = ko.observable('');
+
+    self.loading = ko.observable(false);
+    self.creating = ko.observable(false);
+    self.creatingCredentials = ko.observable(false);
+
+    self.message = ko.observable('');
```

Comment:
Consider using the opp module, and the ChangeMessageMixin: https://github.com/CenterForOpenScience/osf.io/blob/0b8090df9a72d96c2c029382243a7c63e6d17f55/website/static/js/accountSettings.js#L147